saw: init at 0.2.2

terlar opened the pull request:
###### Description of changes

Add package [saw](https://github.com/TylerBrock/saw) - a fast, multi-purpose tool for AWS CloudWatch Logs.

I guess it is a bit similar to the already packaged [awslogs](https://github.com/jorgebastida/awslogs). I have not tried that one yet, so I am not fully aware of the differences.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

terlar committed:
saw: init at 0.2.2

papojari commented:
> I guess it is a bit similar to the already packaged [awslogs](https://github.com/jorgebastida/awslogs). I have not tried that one yet, so I am not fully aware of the differences.

Well, you should.

papojari commented on pkgs/tools/admin/saw/default.nix:
```diff
@@ -0,0 +1,36 @@
+  };
+
+  meta = with lib; {
+    description = "Fast, multi-purpose tool for AWS CloudWatch Logs.";
```
```suggestion
    description = "Fast, multi-purpose tool for AWS CloudWatch Logs";
```

papojari commented:
Result of `nixpkgs-review pr 166956` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>saw</li>
  </ul>
</details>

terlar commented:
So apparently the creator of saw was a dissatisfied user of awslogs, perhaps in the words of the author as well as some commentors:
https://stackoverflow.com/questions/34018931/how-to-view-aws-log-real-time-like-tail-f#answer-50916992

Seems the UX and execution of saw is a bit nicer, and that has been my experience as well as I briefly explored awslogs now.

I guess these tools become less relevant with the updated awscli v2. But they still seem to have some advantages. Will explore this as well.

Addressed your comments.

papojari commented:
It is done, right?

papojari commented:
It is done, right?

terlar commented:
Yes, it is slightly different. A bit nicer UX and potentially faster, but the true bottleneck is CloudWatch itself.

Yes, it is done. Good to go.

jiegec commented:
`testVersion` should be replaced by `testers.testVersion`.
